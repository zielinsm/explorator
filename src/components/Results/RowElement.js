import React from "react";
import { Label, Table } from "semantic-ui-react";

import truncate from "../../helpers/truncate";
import "./RowElement.css";

const RowElement = props => {
  const data = props.row;
  const rowClick = props.rowClick;

  const onClick = function onRowClick(e) {
    const { fingerprint } = data;
    rowClick(e, fingerprint);
  };

  const renderLabel = function renderRelayLabel(status) {
    if (status) {
      return <Label color="green">Running</Label>;
    } else {
      return <Label color="red">Offline</Label>;
    }
  };

  const { name, fingerprint, address, running } = data;
  const truncatedFingerprint = truncate(fingerprint, 15);

  return (
    <Table.Row onClick={onClick} className="RowElement">
      <Table.Cell>{name}</Table.Cell>
      <Table.Cell>{address[0]}</Table.Cell>
      <Table.Cell>{truncatedFingerprint}</Table.Cell>
      <Table.Cell>{renderLabel(running)}</Table.Cell>
    </Table.Row>
  );
};

export default RowElement;
