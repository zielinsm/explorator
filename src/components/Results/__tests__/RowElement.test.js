import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import RowElement from "../RowElement";
import { Table } from "semantic-ui-react";

const mockData = {
  name: "mockRelay",
  address: ["000.000.000.000:9001"],
  fingerprint: "LONGFINGERPRINT",
  running: true
};

describe("RowElement component:", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<RowElement row={mockData} />);
    expect(wrapper.find(".RowElement")).toHaveLength(1);
  });

  it("handles clicks properly", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<RowElement row={mockData} rowClick={onClick} />);
    wrapper.find(".RowElement").simulate("click");
    expect(onClick).toBeCalled();
  });

  it("snapshot matches", () => {
    const tree = renderer.create(<RowElement row={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
