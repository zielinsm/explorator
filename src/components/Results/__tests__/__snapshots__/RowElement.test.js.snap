// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowElement component: snapshot matches 1`] = `
<tr
  className="RowElement"
  onClick={[Function]}
>
  <td
    className=""
  >
    mockRelay
  </td>
  <td
    className=""
  >
    000.000.000.000:9001
  </td>
  <td
    className=""
  >
    LONGFINGERPRINT
  </td>
  <td
    className=""
  >
    <div
      className="ui green label"
      onClick={[Function]}
    >
      Running
    </div>
  </td>
</tr>
`;
